### ISSUE TITLE
Signatures for LexRuntime seem to be invalid

### ISSUE THREAD

**Issue by simcop2387** [pplu/aws-sdk-perl#215]
So when using Lex via a cognito identity pool, there's a signature error because the signature is set for the scope 'runtime.lex' instead of 'lex'.  As a quick attempt to work around this, I added a ```sub _service_for_signature {'lex'}``` to Paws::LexRuntime and changed the corrosponding part of Paws::Net::V4Signature to use it if present.

```perl
package Paws::Net:::V4Signature;
# other code elided ...

    my $sig = Net::Amazon::Signature::V4->new( $self->access_key, $self->secret_key, $self->_region_for_signature, $self->can('_service_for_signature')?$self->_service_for_signature : $self->service );
```

A quick example of the error is below

```perl
#!/usr/bin/env perl

use strict;
use warnings;

use Data::Dumper;
use JSON;
use Paws;
use Paws::Credential::Explicit;

my $creds = Paws::Credential::Explicit->new(access_key => '', secret_key => '');

my $p1 = Paws->new(config => {credentials => $creds, region => 'us-east-1'});

my $cognito = $p1->service("CognitoIdentity");

my $pool_id = 'us-east-1:dbe1d89b-d4ed-459d-ba40-ef8f096b5e4b';

my $ident_id = $cognito->GetId(IdentityPoolId => $pool_id)->IdentityId;

my $cresp = $cognito->GetCredentialsForIdentity(IdentityId => $ident_id)->Credentials;
my $realcreds = Paws::Credential::Explicit->new(access_key=>$cresp->AccessKeyId, secret_key=>$cresp->SecretKey, session_token=>$cresp->SessionToken);

my $service_obj = $p1->service('LexRuntime', credentials => $realcreds, region=>'us-east-1');

my $resp = $service_obj->PostText(BotName=>'BookTrip', BotAlias=>'perlbot',InputText=>"Hi",UserId=>'me');

print Dumper($resp);
```

**Comment by simcop2387** [comment]
I don't know what the impact of setting up that method for signing the credentials is fully.  If you've got any guidance on what would be the preferred way to handle it I'll make a pull request.  You're fine to use the identity pool id above to access/test yourself but it shouldn't end up in any .t files

**Comment by simcop2387** [comment]
Looks like botocore calls this signingName, I'll work on a PR that uses this instead to make a proper fix.

https://github.com/pplu/botocore/blob/bc9ff533b72614bfb6ffeffc251e8e26d6952e91/botocore/data/lex-runtime/2016-11-28/service-2.json

**Comment by pplu** [comment]
Thanks for the information. There are more services in the same state. I'm working on https://github.com/pplu/aws-sdk-perl/tree/feature/endpoint-update to try to fix this for all services. Please make me arrive your work if you can (to not duplicate efforts)

**Comment by pplu** [comment]
JFYI: PR #243 has a fix for this. It will be released with 0.37

**Comment by pplu** [comment]
Paws 0.37 with this change has just been uploaded to CPAN
